services/galley/src/Galley/API/Update.hs

```diff
@@ -424,8 +427,29 @@ postNewOtrMessage usr con cnv val msg = do
         let (toBots, toUsers) = foldr (newMessage usr con (Just cnv) msg now) ([],[]) rs
         pushSome (catMaybes toUsers)
         void . fork $ do
-            gone <- External.deliver toBots
+            dewhitelisted <- getDewhitelisted (map fst toBots)
```

Comment:
How about checking that `toBots` is not empty? Assuming most conversations _do not_ have bots, we'd be saving an RPC for every message sent.